cli/cmd/edges.go

```diff
@@ -260,58 +278,67 @@ func writeEdgesToBuffer(rows []*pb.Edge, w *tabwriter.Writer, options *edgesOpti
 	})
 
 	switch options.outputFormat {
-	case tableOutput:
+	case tableOutput, wideOutput:
 		if len(edgeRows) == 0 {
 			fmt.Fprintln(os.Stderr, "No edges found.")
 			os.Exit(0)
 		}
-		printEdgeTable(edgeRows, w, maxSrcLength, maxDstLength, maxClientLength, maxServerLength, maxMsgLength)
+		printEdgeTable(edgeRows, w, maxSrcLength, maxSrcNamespaceLength, maxDstLength, maxDstNamespaceLength, maxClientLength, maxServerLength, maxMsgLength, options.outputFormat)
 	case jsonOutput:
 		printEdgesJSON(edgeRows, w)
 	}
 }
 
-func printEdgeTable(edgeRows []edgeRow, w *tabwriter.Writer, maxSrcLength, maxDstLength, maxClientLength, maxServerLength, maxMsgLength int) {
+func printEdgeTable(edgeRows []edgeRow, w *tabwriter.Writer, maxSrcLength, maxSrcNamespaceLength, maxDstLength, maxDstNamespaceLength, maxClientLength, maxServerLength, maxMsgLength int, outputFormat string) {
 	srcTemplate := fmt.Sprintf("%%-%ds", maxSrcLength)
 	dstTemplate := fmt.Sprintf("%%-%ds", maxDstLength)
+	srcNamespaceTemplate := fmt.Sprintf("%%-%ds", maxSrcNamespaceLength)
+	dstNamespaceTemplate := fmt.Sprintf("%%-%ds", maxDstNamespaceLength)
+	msgTemplate := fmt.Sprintf("%%-%ds", maxMsgLength)
 	clientTemplate := fmt.Sprintf("%%-%ds", maxClientLength)
 	serverTemplate := fmt.Sprintf("%%-%ds", maxServerLength)
-	msgTemplate := fmt.Sprintf("%%-%ds", maxMsgLength)
 
 	headers := []string{
 		fmt.Sprintf(srcTemplate, srcHeader),
 		fmt.Sprintf(dstTemplate, dstHeader),
-		fmt.Sprintf(clientTemplate, clientHeader),
-		fmt.Sprintf(serverTemplate, serverHeader),
-		fmt.Sprintf(msgTemplate, msgHeader),
+		fmt.Sprintf(srcNamespaceTemplate, srcNamespaceHeader),
+		fmt.Sprintf(dstNamespaceTemplate, dstNamespaceHeader),
+	}
+
+	if outputFormat == wideOutput {
+		headers = append(headers, fmt.Sprintf(clientTemplate, clientHeader), fmt.Sprintf(serverTemplate, serverHeader))
 	}
 
-	headers[len(headers)-1] = headers[len(headers)-1] + "\t" // trailing \t is required to format last column
+	headers = append(headers, fmt.Sprintf(msgTemplate, msgHeader)+"\t")
 
 	fmt.Fprintln(w, strings.Join(headers, "\t"))
 
 	for _, row := range edgeRows {
-		values := make([]interface{}, 0)
-		templateString := fmt.Sprintf("%s\t%s\t%s\t%s\t%s\t\n", srcTemplate, dstTemplate, clientTemplate, serverTemplate, msgTemplate)
-
-		values = append(values, []interface{}{
+		values := []interface{}{
 			row.src,
 			row.dst,
-			row.client,
-			row.server,
-			row.msg,
-		}...)
+			row.srcNamespace,
+			row.dstNamespace,
+		}
+		templateString := fmt.Sprintf("%s\t%s\t%s\t%s\t%s\t\n", srcTemplate, dstTemplate, srcNamespaceTemplate, dstNamespaceTemplate, msgTemplate)
 
-		fmt.Fprintf(w, templateString, values...)
+		if outputFormat == wideOutput {
+			templateString = fmt.Sprintf("%s\t%s\t%s\t%s\t%s\t%s\t%s\t\n", srcTemplate, dstTemplate, srcNamespaceTemplate, dstNamespaceTemplate, clientTemplate, serverTemplate, msgTemplate)
 
+			values = append(values, row.client, row.server)
+		}
+
+		values = append(values, row.msg)
```

Comment:
TIOLI, to parallel how you're incrementally constructing the `values` slice, you could also incrementally build the `templateString` string, rather than needing to fully redefine it inside the if condition. I'm thinking something like this:

```diff
diff --git a/cli/cmd/edges.go b/cli/cmd/edges.go
index d7bf53eb..0a295f0c 100644
--- a/cli/cmd/edges.go
+++ b/cli/cmd/edges.go
@@ -324,14 +324,15 @@ func printEdgeTable(edgeRows []edgeRow, w *tabwriter.Writer, maxSrcLength, maxSr
 			row.srcNamespace,
 			row.dstNamespace,
 		}
-		templateString := fmt.Sprintf("%s\t%s\t%s\t%s\t%s\t\n", srcTemplate, dstTemplate, srcNamespaceTemplate, dstNamespaceTemplate, msgTemplate)
 
-		if outputFormat == wideOutput {
-			templateString = fmt.Sprintf("%s\t%s\t%s\t%s\t%s\t%s\t%s\t\n", srcTemplate, dstTemplate, srcNamespaceTemplate, dstNamespaceTemplate, clientTemplate, serverTemplate, msgTemplate)
+		templateString := fmt.Sprintf("%s\t%s\t%s\t%s\t", srcTemplate, dstTemplate, srcNamespaceTemplate, dstNamespaceTemplate)
 
+		if outputFormat == wideOutput {
+			templateString += fmt.Sprintf("%s\t%s\t", clientTemplate, serverTemplate)
 			values = append(values, row.client, row.server)
 		}
 
+		templateString += fmt.Sprintf("%s\t\n", msgTemplate)
 		values = append(values, row.msg)
 
 		fmt.Fprintf(w, templateString, values...)
```

That makes the two variables feel more similar, to me.

Reply:
I like this approach, it adds consistency.